File: 백준/Silver/10828. 스택/스택.py
import sys


def input():
    return sys.stdin.readline()


n = int(input())
stack = []
for _ in range(n):
    commands = input().split()
    if len(commands) == 2:
        stack.append(int(commands[1]))
    else:
        cmd = commands[0]
        if cmd == 'pop':
            if len(stack) == 0:
                print(-1)
            else:
                print(stack[-1])
                stack.pop()
        elif cmd == 'size':
            print(len(stack))
        elif cmd == 'empty':
            print(1 if len(stack) == 0 else 0)
        elif cmd == 'top':
            if len(stack) == 0:
                print(-1)
            else:
                print(stack[-1])

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n, x;
	string cmd;
	stack<int> s;
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd == "push") {
			cin >> x;
			s.push(x);
			continue;
		}
		else if (cmd == "pop") {
			if (s.empty()) cout << -1;
			else {
				cout << s.top();
				s.pop();
			}
		}
		else if (cmd == "size") {
			cout << s.size();
		}
		else if (cmd == "empty") {
			cout << s.empty() ? 1 : 0;
		}
		else if (cmd == "top") {
			if (s.empty()) cout << -1;
			else cout << s.top();
		}
		cout << '\n';
	}
}